external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -67,21 +71,28 @@ object SchemaConverters {
       case ENUM => SchemaType(StringType, nullable = false)
 
       case RECORD =>
+        if (existingRecordNames.contains(avroSchema.getFullName)) {
```

Comment:
Another approach is to check the whole json string schema(`avroSchema.toString`) here. But it seems overkill. Avro requires the full name of record to be unique.